cradle-core/src/main/java/com/exactpro/cradle/serialization/MessageSerializer.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2021-2021 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.cradle.serialization;
+
+import com.exactpro.cradle.messages.StoredMessage;
+import com.exactpro.cradle.messages.StoredMessageId;
+import com.exactpro.cradle.messages.StoredMessageMetadata;
+import com.exactpro.cradle.testevents.StoredTestEventId;
+
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Map;
+
+import static com.exactpro.cradle.serialization.Serialization.MessageBatchConst.*;
+import static com.exactpro.cradle.serialization.SerializationUtils.printBody;
+import static com.exactpro.cradle.serialization.SerializationUtils.printInstant;
+import static com.exactpro.cradle.serialization.SerializationUtils.printShortString;
+import static com.exactpro.cradle.serialization.SerializationUtils.printString;
+
+public class MessageSerializer {
+
+	public byte[] serializeBatch(Collection<StoredMessage> batch) throws SerializationException {
+		SerializationBatchSizes messageBatchSizes = calculateMessageBatchSize(batch);
```

Comment:
Please use `calculateMessageBatchSize()` method in calculating current batch size in `StoredMessageBatch` where it checks if message being added won't overload the batch.